WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingViewModel.kt

```diff
@@ -196,4 +200,11 @@ class PrepublishingViewModel @Inject constructor() : ViewModel() {
         val formattedTags = TextUtils.join(",", tags)
         return StringEscapeUtils.unescapeHtml4(formattedTags)
     }
+
+    /**
+     * Fetches the tags so that they will be available when the Tags action is clicked
+     */
+    private fun fetchTags() {
+        dispatcher.dispatch(TaxonomyActionBuilder.newFetchTagsAction(site))
```

Comment:
Don't we need to check if the action has completed? Or what happens when the action doesn't complete and the user clicks on the Tags action?

Reply:
Thanks for pointing this out @malinajirka 🙇 So to answer the first part of your question, we don't check if the action completes because it seems like a fire & forget approach, especially since we are on a screen that doesn't even need these tags. I am doing this by following the approach used in the `EditPostSettingsFragment`
https://github.com/wordpress-mobile/WordPress-Android/blob/31ef563ac86efeb9a78ccf914fa30a5fd4359f4e/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java#L579

This expects that by the time the user tries to edit the tags this operation would be completed in the background and they would see a preloaded list of previously added tags.

For the second part of your question, if the user clicks on the tags action and the action didn't complete then they won't see the preloaded list but it seems once they have been fetched once all is well. In terms of errors,  I simply logged it as seen here 5c8e893

Let me know what you think. Thanks 🙏